### ISSUE TITLE
Behavioural cloning from replay files to bootstrap policy weights

### ISSUE THREAD

**Issue by espenhk** [espenhk/gamer-ai#23]
## Summary

Add a **behavioural cloning (BC)** pre-training script that reads existing `.Replay.Gbx` files (already present in `replays/`), reconstructs state-action pairs, and uses supervised learning to initialise `WeightedLinearPolicy` or `NeuralNetPolicy` weights before RL training begins.

## Motivation

All current training methods start from random weights and learn purely from self-play, which requires many episodes just to discover that accelerating is good. A single human replay of `a03` contains thousands of high-quality state-action pairs that could:

1. **Bootstrap cold-start**: replace the current probe + random cold-start with a policy that already drives correctly
2. **Reduce sample complexity**: RL fine-tuning from a BC initialisation converges 5–50× faster than from scratch
3. **Establish meaningful baselines**: a BC policy alone (without further RL) shows how close "imitate the human" gets

The replay file `replays/a03_centerline.Replay.Gbx` is already used by `build_centerline.py` (via `pygbx`), so the parsing infrastructure exists.

### New script: `pretrain_bc.py`

```
python pretrain_bc.py replays/a03_centerline.Replay.Gbx \
    --policy linear     # or "neural_net"
    --output experiments/a03_centerline/bc_init/policy_weights.yaml
    --epochs 100
    --lr 0.01
```

### Data extraction

The `pygbx` library can deserialise `.Gbx` replay files. Each replay tick provides:

- **Position** (x, y, z) → use `Centerline.project()` to compute `lateral_offset`, `vertical_offset`, `track_progress`
- **Velocity** → `speed_ms`
- **Rotation** (quaternion) → `yaw_error`, `pitch`, `roll` via `Quat` in `utils.py`
- **Input state** (steer int, accel bool, brake bool) → the demonstration action

Derived observations (wheel contacts, angular velocity) may not be available in all replay formats — these can be zeroed or estimated from finite differences.

### Supervised learning objective

For `WeightedLinearPolicy`:

```
# Steer: MSE regression
L_steer = mean( (steer_weights @ norm_obs - steer_demo)^2 )

# Accel: binary cross-entropy
L_accel = mean( -[a_demo * log(σ(w_a @ obs)) + (1-a_demo) * log(1-σ(...))] )

# Brake: same
L_brake = ...

L_total = L_steer + L_accel + L_brake
```

For `NeuralNetPolicy`: identical loss applied to network output layer, with backprop through hidden layers.

### Integration with existing training

After `pretrain_bc.py` writes `policy_weights.yaml`, the existing `main.py` flow loads it via `WeightedLinearPolicy._load_or_init()` — zero code changes to main training loop required.

Optionally, add a `--bc_pretrain` flag to `main.py`:

```yaml
# training_params.yaml
bc_pretrain_replay: replays/a03_centerline.Replay.Gbx   # or null
bc_epochs: 50
```

## Files to change

| File | Change |
|------|--------|
| `pretrain_bc.py` | New script (~200 lines) |
| `main.py` | Optional: read `bc_pretrain_replay` before cold-start; call `pretrain_bc` inline |
| `config/training_params.yaml` | Document `bc_pretrain_replay`, `bc_epochs` |
| `utils.py` | May need `StateData.from_gbx_sample()` helper |
| `tests/test_bc.py` | Unit test: BC on synthetic trajectory reduces steer loss |

## Data Considerations

- **Steer mapping**: replay stores steer as integer ±65536; normalise by `STEER_SCALE` (constants.py)
- **Frame rate**: GBX replays at 100 Hz; TMInterface env runs at ~100 Hz; 1:1 correspondence
- **Missing features**: `turning_rate` (obs[7]) is the steer from the previous tick — available directly from replay. `wheel_N_contact` and `angular_vel` may need estimation.
- **Lookahead**: computed from `Centerline.project()` at positions along the replay path — fully reconstructable

## Acceptance Criteria

- [ ] `pretrain_bc.py` parses a `.Replay.Gbx` file and extracts (obs, action) pairs
- [ ] Supervised training loop runs for `--epochs` passes, reporting train loss per epoch
- [ ] Output `policy_weights.yaml` is valid input for `WeightedLinearPolicy._load_or_init()`
- [ ] BC-initialised policy drives the track directionally correct on first episode (qualitative check)
- [ ] Works for both `linear` and `neural_net` policy types
- [ ] `main.py` respects optional `bc_pretrain_replay` config key before starting cold-start

## References

- Ross & Bagnell (2010). *Efficient Reductions for Imitation Learning.*
- Rajeswaran et al. (2017). *Learning Complex Dexterous Manipulation with Deep Reinforcement Learning and Demonstrations.* arXiv:1709.10087
